tools/find-inactive-tsc.mjs

```diff
@@ -0,0 +1,260 @@
+  const readmeText = readline.createInterface({
+    input: fs.createReadStream(new URL('../README.md', import.meta.url)),
+    crlfDelay: Infinity,
+  });
+  let fileContents = '';
+  let inTscSection = false;
+  let inTscEmeritusSection = false;
+  let memberFirstLine = '';
+  const textToMove = [];
+  let moveToInactive = false;
+  for await (const line of readmeText) {
+    // If we've been processing TSC emeriti and we reach the end of
+    // the list, print out the remaining entries to be moved because they come
+    // alphabetically after the last item.
+    if (inTscEmeritusSection && line === '' &&
+        fileContents.endsWith('>\n')) {
+      while (textToMove.length) {
+        fileContents += textToMove.pop();
+      }
+    }
+
+    // If we've found the TSC heading already, stop processing at the
+    // next heading.
+    if (line.startsWith('#')) {
+      inTscSection = false;
+      inTscEmeritusSection = false;
+    }
+
+    const isTsc = inTscSection && line.length;
+    const isTscEmeritus = inTscEmeritusSection && line.length;
+
+    if (line === '### TSC (Technical Steering Committee)') {
+      inTscSection = true;
+    }
+    if (line === '### TSC emeriti') {
+      inTscEmeritusSection = true;
+    }
+
+    if (isTsc) {
+      if (line.startsWith('* ')) {
+        memberFirstLine = line;
+        const match = line.match(/^\* \[([^\]]+)/);
+        if (match && peopleToMove.includes(match[1])) {
+          moveToInactive = true;
+        }
+      } else if (line.startsWith('  **')) {
+        if (moveToInactive) {
+          textToMove.push(`${memberFirstLine}\n${line}\n`);
+          moveToInactive = false;
+        } else {
+          fileContents += `${memberFirstLine}\n${line}\n`;
+        }
+      } else {
+        fileContents += `${line}\n`;
+      }
+    }
+
+    if (isTscEmeritus) {
+      if (line.startsWith('* ')) {
+        memberFirstLine = line;
+      } else if (line.startsWith('  **')) {
+        const currentLine = `${memberFirstLine}\n${line}\n`;
+        // If textToMove is empty, this still works because when undefined is
+        // used in a comparison with <, the result is always false.
+        while (textToMove[0] < currentLine) {
+          fileContents += textToMove.shift();
+        }
+        fileContents += currentLine;
+      } else {
+        fileContents += `${line}\n`;
+      }
+    }
+
+    if (!isTsc && !isTscEmeritus) {
+      fileContents += `${line}\n`;
+    }
+  }
+
+  return fileContents;
+}
+
+// Get current TSC members, then get TSC members at start of period. Only check
+// TSC members who are on both lists. This way, we don't flag someone who has
+// only been on the TSC for a week and therefore hasn't attended any meetings.
+const tscMembersAtEnd = await getTscFromReadme();
+
+await runGitCommand(`git checkout 'HEAD@{${SINCE}}' -- README.md`);
+const tscMembersAtStart = await getTscFromReadme();
+await runGitCommand('git reset HEAD README.md');
```

Comment:
Is the reset necessary here? It looks to me that the checkout next line should be sufficient.
```suggestion
```

Reply:
Based on running the commands manually, it seems the reset is necessary. Without it, the subsequent checkout does not restore the file to the most recent version.